src/library_html5.js

```diff
@@ -935,17 +935,32 @@ var LibraryHTML5 = {
 
   $fillOrientationChangeEventData__deps: ['$screenOrientation'],
   $fillOrientationChangeEventData: (eventStruct) => {
-    var orientations  = ["portrait-primary", "portrait-secondary", "landscape-primary", "landscape-secondary"];
-    var orientations2 = ["portrait",         "portrait",           "landscape",         "landscape"];
-
-    var orientationString = screenOrientation();
-    var orientation = orientations.indexOf(orientationString);
-    if (orientation == -1) {
-      orientation = orientations2.indexOf(orientationString);
+    // OrientationType enum
+    var orientationsType1 = ["portrait-primary", "portrait-secondary", "landscape-primary", "landscape-secondary"];
+    // alternative selection from OrientationLockType enum
+    var orientationsType2 = ["portrait",         "portrait",           "landscape",         "landscape"];
+
+    var orientationIndex = -1;
+    var orientationAngle =  0;
+    var screenOrientObj  = screenOrientation();
+    if (screenOrientObj) {
+      orientationIndex = orientationsType1.indexOf(screenOrientObj.type);
+      if (orientationIndex == -1) {
+        orientationIndex = orientationsType2.indexOf(screenOrientObj.type);
+      }
+      if (orientationIndex != -1) {
+        orientationIndex = 1 << orientationIndex;
+      }
+      orientationAngle = screenOrientObj.angle;
+    } else {
```

Comment:
How about we wrap this whole `else` block in `MIN_SAFARI_VERSION != TARGET_NOT_SUPPORTED` to make the comment below explict?

Reply:
I'll verify the behaviour on older Safari versions and add this.

Reply:
How does the `TARGET_NOT_SUPPORTED` work?

I'm just looking at this on various browsers and OSes and the only one that stands out is Safari, but 16.4 onwards has full support on desktop and mobile (whereas Safari mobile can only call `emscripten_get_orientation_status()` and get the angle through `window.orientation`).

Reply:
Users can build with `-sMIN_SAFARI_VERSION=-1` to completely disable safari support.  In that case `TARGET_NOT_SUPPORTED` will be the value of `MIN_SAFARI_VERSION` and you can get any explicit safari hacks on that.